### PR TITLE
feat(types,rpc,node)!: refactor Share to work for parity and data

### PR DESCRIPTION
**Pull request by zvolin**

### CODE REVIEW

**Comment by distractedm1nd** [comment]
image

File: types/src/rsmt2d.rs [hunk 1]
```diff
@@ -476,6 +466,10 @@ pub(crate) fn is_ods_square(row: u16, column: u16, square_width: u16) -> bool {
     row < ods_width && column < ods_width
 }
 
+fn flatten_index(row: u16, col: u16, square_width: u16) -> usize {
```
Comment (oblique) [inline types/src/rsmt2d.rs:469]: nice name!

**Review by oblique — APPROVED** [review]
nice!

**Review by fl0rek — APPROVED** [review]
:guitar:

**Pull request merged**